### ISSUE TITLE
Move index.html from src to dist folder

### ISSUE THREAD

**Issue by alaksandarjesus** [evanw/esbuild#1771]
I am learning esbuild and want to test with a simple project. I am struck on moving the src/index.html to dist/index.html

Is there any loader available that runs or moves only once, and watches for changes in html files.

[https://github.com/alaksandarjesus/esbuilder] (github link to project setup)

**Comment by hyrious** [comment]
Hey, just write a simple script or call [cpy](https://github.com/sindresorhus/cpy-cli) command after your build. Don't stuck yourself in one tool for all.

For example, [here](https://github.com/hyrious/esbuild-repl/blob/main/scripts/dev.ts)'s my dev script that simply watchs source folder and builds everything out, which also triggers a hot-refresh in the browser, it even includes an SSR index.html output.

**Comment by leuvi** [comment]
I made a simple esbuild build demo, you can refer to [this](https://github.com/leuvi/anole/blob/main/config/build.js)

**Comment by vzakharov-rxnt** [comment]
But it's more than simply copying. It also needs to update that HTML file to include content hashed scripts and CSS links.
How would you do that?

**Comment by Discountrobot** [comment]
@vzakharov-rxnt what you are describing not supported at the moment. https://github.com/evanw/esbuild/issues/1995

**Comment by vzakharov-rxnt** [comment]
@Discountrobot I see. But I think these days building js/css is only part of the issue. Any real production use case cannot survive on this. Esbuild can build react jsx but cannot build a create react app boilerplate, that doesn't make much sense.

It would be great to see html as first class support, here on in another repo as a plugin. I don't think esbuild can make the intended impact in the world without it. I understand there are many open source CLIs that integrate esbuild under the hood, reality is such that webpack/esbuild-loader combo beats them all by at least 2x in terms of build speed. So all that effort is wasted, for now.

**Comment by Discountrobot** [comment]
@vzakharov-rxnt while i cannot speak on the behalf of the maintainers of this project, it does seem that direction is pretty clear - do bundling well and leave the rest to be sorted in plugins.

i personally disagree extending the esbuild toolchain to support features outside bundling, (like the [`serve`](https://esbuild.github.io/api/#serve) feature, which should have been a plugin akin to the [devServer](https://webpack.js.org/configuration/dev-server/)).

Maybe you want the html templating to use [pug.js](https://pugjs.org/api/getting-started.html), while i'd like [ejs](https://ejs.co/) support. Or maybe you're using esbuild to bundle your Node.js backend and do not require templating at all.

The javascript community is finding itself in dependency purgatory at the moment - and if you are part of an organisation that value security aspects of your projects, you should select few chosen dependencies.

I'd recommend, as you mention, to go with one of the existing of tooling libraries like [vite](https://vitejs.dev/) and webpack plugins that use esbuild in conjunction with other existing plugins.

or write the esbuild plugin required by your organization, there are already a few [for html templating ](https://www.npmjs.com/search?q=esbuild%20html)

**Comment by evanw** [comment]
I'm going to close this issue since this there's no more activity in this discussion, it's a non-goal for esbuild to replicate "create react app" boilerplate, and HTML entry points are already covered by #31.
